consensus/src/dag/order_rule.rs

```diff
@@ -59,12 +60,37 @@ impl OrderRule {
                     if let Some(anchor) = maybe_anchor {
                         dag.write()
                             .reachable_mut(&anchor, None)
-                            .for_each(|node_status| node_status.mark_as_ordered());
+                            .for_each(|node_status| {
+                                debug!(
+                                    "Marking node as ordered from CommitEvent: {}",
+                                    node_status.as_node().id()
+                                );
+                                node_status.mark_as_ordered();
+                            });
                     }
                 }
                 anchor_election.update_reputation(event);
             }
+        } else if lowest_unordered_anchor_round > 1 && !dag.read().is_empty() {
```

Comment:
why do we need this branch? can't we just read the commit events all the time?

Reply:
I don't want to rely on events in RoundRobin mode (in case there are any bugs).

Reply:
doesn't make much sense since it requires 2f+1 to write commit events?

Reply:
reading can be a problem?

Reply:
okay, let's trust the events. i will try to clean up later.